src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2258,10 +2308,12 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 		 * LIMIT and ORDER BY clauses as described below:
 		 *      (1) Creating a new group by clause during aggregate mutation, or
 		 *      (2) Distinct clause is not pushed down
+		 *		(3) Window function needs to be processed before limit
```

Comment:
We could skip adding `(3)` since we already skip this block when there are non pushdownable window functions